static/js/shared/data_fetcher.test.ts

```diff
@@ -160,6 +161,284 @@ test("fetch stats data", () => {
   });
 });
 
+test("fetch stats data where latest date with data for all stat vars is not the latest date", () => {
+  mockedAxios.get.mockImplementation((url: string) => {
+    if (url === "/api/stats/Count_Person?&dcid=geoId/05&dcid=geoId/06") {
+      return Promise.resolve({
+        data: {
+          "geoId/05": {
+            data: {
+              "2011": 21000,
+              "2012": 22000,
+            },
+            placeName: "Arkansas",
+            provenanceDomain: "source1",
+          },
+          "geoId/06": {
+            data: {
+              "2011": 31000,
+              "2013": 32000,
+            },
+            placeName: "California",
+            provenanceDomain: "source2",
+          },
+        },
+      });
+    }
+  });
+
+  return fetchStatsData(["geoId/05", "geoId/06"], ["Count_Person"]).then(
+    (data) => {
+      expect(data).toEqual({
+        data: {
+          Count_Person: {
```

Comment:
Can we factor out this data object so that we don't have to repeat the entire chunk?